### ISSUE TITLE
Database connection checks at startup

### ISSUE THREAD

**Issue by matheusgomes28** [matheusgomes28/urchin#51]
It seems like the App starts up **even** when the database connection isn't right.

The following line does not return an error when the connection isn't valid in `cmd/urchin/main.go`:

```
db_connection, err := database.MakeSqlConnection(
  app_settings.DatabaseUser,
  app_settings.DatabasePassword,
  app_settings.DatabaseAddress,
  app_settings.DatabasePort,
  app_settings.DatabaseName,
)
```

Make it so we check the connection and fail fast if the DB connection is invalid!

**Comment by Milad75Rasouli** [comment]
it's my idea to fix the issue: 
```
	err = db.Ping()
	if err != nil {
		return nil, err
	}
```
what do you say?

**Comment by matheusgomes28** [comment]
Hey @Milad75Rasouli , the plan sounds good!

If you push the fix, I can review it after work. Please don't forget to run the project and make sure it still works as intended 😄

**Comment by Milad75Rasouli** [comment]
@matheusgomes28  TBH, you've made the running project flow a bit difficult. either your scripts for running the project or the Dockerfile and docker-compose files need to be improved. IMO, there are a lot of thing to do with them, however I'll do my change and run the project for sure. A solid document to explain new contributors how to run the project is required.

**Comment by matheusgomes28** [comment]
@Milad75Rasouli I agree, I'm gathering some feedback to figure out how to make the flow easier.

I definitely need more instructions on the `README.md` for sure, for starters. Feel free to raise any issues with potential feedback on improving the workflow too, in case you have suggestions.

**Comment by AlDu2407** [comment]
> @matheusgomes28 TBH, you've made the running project flow a bit difficult. either your scripts for running the project or the Dockerfile and docker-compose files need to be improved. IMO, there are a lot of thing to do with them, however I'll do my change and run the project for sure. A solid document to explain new contributors how to run the project is required.

I'm preparing a PR with additional documentation and an easier setup for starting the development process. Will create it by the next weekend and it should make the onboarding process easier. You'll have to wait for it until next weekend though. :)

**Comment by matheusgomes28** [comment]
closed with #53
